MSAL/test/integration/native_auth/end_to_end/otp_code_retriever/MSALNativeAuthEmailCodeRetriever.swift

```diff
@@ -76,7 +84,19 @@ class MSALNativeAuthEmailCodeRetriever: XCTestCase {
             }
             if dataDictionary.count > 0 {
                 dataDictionary.sort(by: {($0["id"] as? Int ?? 0) > ($1["id"] as? Int ?? 0)})
-                return dataDictionary.first?["id"] as? Int
+                if let emailDateString = dataDictionary.first?["date"] as? String,
+                   let emailDate = dateFormatter.date(from: emailDateString) {
+                    let currentDate = Date()
+                    // Email should be newer than 5 seconds otherwise it could be from previous test
+                    // This retry will help with the delay in receiving the emails
+                    if currentDate.timeIntervalSince1970 - emailDate.timeIntervalSince1970  < secondsToWaitForEmail {
+                        print ("Email is ok, last receive date: \(emailDate) current date: \(currentDate)")
+                        return dataDictionary.first?["id"] as? Int
+                    } else {
+                        print ("Email is older, last receive date: \(emailDate) current date: \(currentDate)")
```

Comment:
older than what?
Do we need to print this info or they're just for debugging purpose?

Reply:
Changed the text to be more clear why we retry. We should have these logs as they are helpful in the pipeline to see very fast if the service is not sending emails or the API is not replying properly. If there is no log the API is not replying. If the service doesn't send we will get 3 times "Email is from previous tests"